mmf/datasets/mmf_dataset_builder.py

```diff
@@ -119,6 +123,49 @@ def _download_requirement(
 
     def load(self, config, dataset_type, *args, **kwargs):
         self.config = config
+
+        new_config = config
+        split_dataset_from_train = self.config.split_train
+        if split_dataset_from_train:
+            split_range = self._calculate_split_for_dataset_type(dataset_type)
+            new_config = self._modify_dataset_config(config)
+
+        annotations = self._read_annotations(new_config, dataset_type)
+        if annotations is None:
+            return None
+
+        datasets = []
+        for imdb_idx in range(len(annotations)):
+            dataset_class = self.dataset_class
+            dataset = dataset_class(new_config, dataset_type, imdb_idx)
+            datasets.append(dataset)
+
+        dataset = MMFConcatDataset(datasets)
+        if split_dataset_from_train:
+            start, end = split_range
+            dataset_length = len(dataset)
+            start, end = round(start * dataset_length), round(end * dataset_length)
+            seed = new_config.split_train.seed
+            indices = np.random.RandomState(seed).permutation(dataset_length)[start:end]
```

Comment:
Use torch to generate indices rather than np?

Reply:
https://github.com/pytorch/pytorch/issues/17079 
It seems like the numpy interface is much cleaner compared to torch's.
Is there a reason that we prefer to use torch than numpy?

Reply:
Yeah, one extra thing for user to seed. We try to rely on torch as much as possible instead of numpy.